src/plugins/minimal_scene/MinimalSceneRhi.hh

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_GUI_PLUGINS_MINIMALSCENE_MINIMALSCENERHI_HH_
+#define IGNITION_GUI_PLUGINS_MINIMALSCENE_MINIMALSCENERHI_HH_
+
+#include "ignition/gui/Plugin.hh"
+#include "ignition/rendering/RenderTypes.hh"
+
+#include <QSGTexture>
+#include <QSize>
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  /// \brief Render interface class to handle OpenGL / Metal compatibility
+  /// of camera textures in IgnRenderer
+  //
+  /// Each supported graphics API must implement this interface
+  /// to provide access to the underlying render system's texture.
+  class IgnCameraTextureRhi
+  {
+    /// \brief Destructor
+    public: virtual ~IgnCameraTextureRhi();
+
+    /// \brief Update the texture for a camera
+    /// \param[in] _camera Pointer to the camera providing the texture
+    public: virtual void Update(rendering::CameraPtr _camera) = 0;
+
+    /// \brief Get the graphics API texture Id
+    /// \param[out] _texturePtr Pointer to a texture Id
+    public: virtual void TextureId(void* _texturePtr) = 0;
+  };
+
+  /// \brief Ign-rendering renderer.
+  class IgnRenderer;
+  class RenderSync;
+
+  /// \brief Render interface class to handle OpenGL / Metal compatibility
+  /// in RenderThread
+  class RenderThreadRhi
+  {
+    /// \brief Destructor
+    public: virtual ~RenderThreadRhi();
+
+    /// \brief Offscreen surface to render to
+    public: virtual QOffscreenSurface *Surface() const;
+
+    /// \brief Set the offscreen surface to render to
+    //
+    /// \param[in] _surface Off-screen surface format
+    public: virtual void SetSurface(QOffscreenSurface *_surface);
+
+    /// \brief OpenGL context to be passed to the render engine
+    public: virtual QOpenGLContext *Context() const;
+
+    /// \brief Set the OpenGL context to be passed to the render engine
+    //
+    /// \param[in] _surface OpenGL context
```

Comment:
```suggestion
    /// \param[in] _context OpenGL context
```

Reply:
@ahcorde thanks, I'll make these changes.

There's something wrong with setting the OpenGL context when running on Linux which I'm looking into - it's one of the reasons the CI test is failing. I'm able to replicate it on an Ubuntu VM using a software OpenGL implementation but can't see where I've deviated from the treatment in main.

On the plus side I've figured out that I can remove the single threaded Qt scene graph restriction for Metal.

The other CI failure is related to a code check for a casting operation - I've changed the code to use the reinterpret_cast but for some reason the CI is still complaining and using the previous version of the source?